packages/asset-swapper/src/quote_consumers/plp_registry.ts

```diff
@@ -0,0 +1,77 @@
+import { ContractFunctionObj } from '@0x/contract-wrappers';
+
+import { constants } from '../constants';
+
+export interface RegistryContract {
+    getPoolForMarket(marketA: string, marketB: string): ContractFunctionObj<string>;
+}
+
+interface CacheValue {
+    value?: string;
+    cachedAt: number;
+}
+
+/**
+ * A wrapper around the PLP Registry that includes a caching layer to minimize latency.
+ */
+export class PLPRegistry {
+    private readonly _contract: RegistryContract;
+    private readonly _cache: { [key: string]: CacheValue };
+
+    /**
+     * Instantiates a new PLPRegistry.
+     * @param contract an instance of IPLPRouterContract
+     */
+    constructor(contract: RegistryContract) {
```

Comment:
Any thoughts on making this more generic so it can adapt any on-chain call to be cached for a time period?

Similar concept with protocol fee multiplier (150,000) where we had discussed checking this every so often.

Reply:
Here is what I came up with: https://gist.github.com/PirosB3/74cb65c65f110927c773e98f6150a7ba

The first file generic_cache_wrapper.ts contains the generic “abstract” classes. The second file plp_registry_cache_wrapper.ts contains an example implementation of the cache wrapper for the PLP Registry. I also added a third file sample_execution.ts which shows how you would “call” the implemented PLP Registry cache wrapper

NOTE: I don’t know TypeScript too well, so some things may be obviously wrong